src/haskell/parser/src/KoreParserImpl.hs

```diff
@@ -0,0 +1,339 @@
+module KoreParserImpl where
+
+import           KoreAST
+import           KoreLexeme
+
+import           Control.Applicative              ((<|>))
+import           Control.Monad                    (void, when)
+
+import           Data.Attoparsec.ByteString.Char8 (Parser)
+import qualified Data.Attoparsec.ByteString.Char8 as Parser
+import           Data.Maybe                       (isJust)
+
+sortVariableParser :: IsMeta a => a -> Parser (SortVariable a)
+sortVariableParser x = SortVariable <$> idParser x
+
+unifiedSortVariableParser :: Parser UnifiedSortVariable
+unifiedSortVariableParser = do
+    c <- Parser.peekChar'
+    if c == '#'
+        then MetaSortVariable <$> sortVariableParser Meta
+        else ObjectSortVariable <$> sortVariableParser Object
+
+sortVariableListParser :: Parser [UnifiedSortVariable]
+sortVariableListParser = Parser.sepBy unifiedSortVariableParser commaParser
+
+sortParser :: IsMeta a => a -> Parser (Sort a)
+sortParser x = do
+    identifier <- idParser x
+    c <- Parser.peekChar
+    case c of
+        Just '{' -> actualSortParser identifier
+        _        -> return (SortVariableSort $ SortVariable identifier)
+  where
+    actualSortParser identifier = do
+        sorts <- inCurlyBracesParser (sortListParser x)
+        when (metaType x == MetaType) (checkMetaSort identifier sorts)
```

Comment:
Isn't illegal for a non-meta sort to start with `#`?

Reply:
a sort is an identifier, so this check is done in the identifier parser